plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt

```diff
@@ -177,5 +178,6 @@ fun TypeEmbedding.isInheritorOfCollectionTypeNamed(name: String): Boolean =
 val TypeEmbedding.isCollectionInheritor
     get() = isInheritorOfCollectionTypeNamed("Collection")
 
-fun TypeEmbedding.subTypeInvariant() = SubTypeInvariantEmbedding(this)
-
+fun TypeEmbedding.subTypeInvariant(): SubTypeInvariantEmbedding? = equals(buildType { nullableAny() }).ifFalse {
+    SubTypeInvariantEmbedding(this)
+}
```

Comment:
Is this special case really worth it?  I guess it reduces the number of inhales we do, but it may be hard to find why nullable any behaves differently.  Perhaps with the type decomposition redesign we can do something nicer.

Reply:
Nah, I think it has some sense to introduce this optimization eventually. Particularly because it can enable further optimizations (like `.simplified`). But I don't think it should be done like this anyway.